### PR TITLE
azure smp nit-picks

### PR DESCRIPTION
**Pull request by tuffacton**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Quality of life additions to what has already been done by Anmol.
* @nikunj-badjatya or someone else from CCM, 90% looks fantastic but please give the section regarding secrets near the end one final review. That is where I had the most confusion so I hope I was able to clear that up but it might need some slight improvements.

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 1]
```diff
@@ -365,18 +374,25 @@ kubectl edit secret smtp-secret -n <namespace>
...
 ## Next steps
 
-- [Kubernetes connector setup](https://developer.harness.io/docs/cloud-cost-management/get-started/onboarding-guide/set-up-cost-visibility-for-kubernetes#create-ccm-connector)
-- [Azure connector setup](https://developer.harness.io/docs/cloud-cost-management/get-started/onboarding-guide/set-up-cost-visibility-for-azure)
+At this point, your "staging area" for future Microsoft Azure CUR reports is ready and we can follow our standard methodologies for onboarding these assets.
+
+**Figure: Azure CCM Self-Managed Documentation Next Steps**
+<DocImage path={require('./static/azure-smp-arch-next-steps.jpeg')} title="Click to view full size image" />
+
+- Use [Azure connector setup](https://developer.harness.io/docs/cloud-cost-management/get-started/onboarding-guide/set-up-cost-visibility-for-azure) to set up Azure CURs and push them to our configured destination storage account.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:396]: Azure Exports

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 2]
```diff
@@ -354,6 +351,18 @@ expiration = 0
...
+- nextgen-ce
+
+```
+kubectl edit secret nextgen-ce -n <namespace>
+```
+
+```
+AZURE_APP_CLIENT_SECRET: "<base64 of clientSecret>"
+```
+After these secrets are implemented, please `kubectl delete` the `nextgen` pods for this release to start a newly configured pod.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:365]: nextgen-ce pods

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 3]
```diff
@@ -365,18 +374,25 @@ kubectl edit secret smtp-secret -n <namespace>
...
-SMTP_PORT: <SMTP_PORT>
-SMTP_USE_SSL: <SMTP_USE_SSL>
-SMTP_USERNAME: <SMTP_USERNAME>
+SMTP_HOST: <base64 of SMTP_HOST>
+SMTP_PASSWORD: <base64 of SMTP_PASSWORD>
+SMTP_PORT: <base64 of SMTP_PORT>
+SMTP_USE_SSL: <base64 of SMTP_USE_SSL>
+SMTP_USERNAME: <base64 of SMTP_USERNAME>
 ```
+After these secrets are implemented, please `kubectl delete` any pods related to `platform-service` for this release to start newly configured pods.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:383]: Also delete pods for batch-processing.

For your reference batch-processing also use SMTP secrets to send Perspective Reports:   https://github.com/harness/harness-core/blob/61d746ecac88cb82894e6c2f0d772e95dcdb3bef/batch-processing/chart/templates/statefulset.yaml#L65

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 4]
```diff
@@ -10,31 +10,43 @@ redirect_from:
...
-<DocImage path={require('./static/azure-smp-arch.png')} width="50%" height="50%" title="Click to view full size image" />
+<DocImage path={require('./static/azure-smp-arch.png')} title="Click to view full size image" />
 
-The flow at a high level works very similar to our saas enviornment. We support syncing data from multiple exports residing in different source storage accounts into CCM via a destination storage account(staging).
+The flow at a high level works very similar to our SaaS enviornment. We support syncing data from multiple Azure account exports residing in different source storage accounts into CCM via a destination storage account (staging).
 
-You need to perform the following tasks to set up CCM for Azure. For Step 1, 2 and 3 Sign in to your [Azure Portal](https://portal.azure.com/#home): 
+The goal of this particular documentation page is to set up an Azure Destination Storage Account and Application Registration, then configue an existing Harness SMP deployment to utilize these components to act as a "staging area" for future Azure CURs to be processed and made available for our Harness CCM capabilities.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:17]: Azure Exports not CURs.

Comment (iamanmolm) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:17]: Also we can move this maybe to the top of the page.

**Review by nikunj-badjatya — DISMISSED** [review]
lgtm

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md (hunk 4 above)
Comment (nikunj-badjatya) [reply docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:17]: Azure Billing Exports.

**Review by iamanmolm — DISMISSED** [review]

**Comment by tuffacton** [comment]
All comments from @nikunj-badjatya and @iamanmolm should be added in new commits, lets get this PR through so we can re-send the link

**Comment by iamanmolm** [comment]
Resolved all the addressed comments and approved the PR.

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 5]
```diff
@@ -10,31 +10,43 @@ redirect_from:
 This topic walks you through the steps required to set up CCM for Azure in a self-managed platform.
 
 **Figure: Azure CCM Self-Managed Enterprise Edition architecture diagram**
-<DocImage path={require('./static/azure-smp-arch.png')} width="50%" height="50%" title="Click to view full size image" />
+<DocImage path={require('./static/azure-smp-arch.png')} title="Click to view full size image" />
 
-The flow at a high level works very similar to our saas enviornment. We support syncing data from multiple exports residing in different source storage accounts into CCM via a destination storage account(staging).
+The flow at a high level works very similar to our SaaS enviornment. We support syncing data from multiple Azure account exports residing in different source storage accounts into CCM via a destination storage account (staging).
```
Comment (brian-f-harness) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:15]:
```suggestion
The flow at a high level works similarly to our SaaS environment. We support syncing data from multiple Azure account exports residing in different source storage accounts into CCM via a destination storage account (staging).
```

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 6]
```diff
@@ -10,31 +10,43 @@ redirect_from:
...
 
-You need to perform the following tasks to set up CCM for Azure. For Step 1, 2 and 3 Sign in to your [Azure Portal](https://portal.azure.com/#home): 
+The goal of this particular documentation page is to set up an Azure Destination Storage Account and Application Registration, then configue an existing Harness SMP deployment to utilize these components to act as a "staging area" for future Azure Billing Exports to be processed and made available for our Harness CCM capabilities.
```
Comment (brian-f-harness) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:17]:
```suggestion
The goal of this particular documentation page is to set up an Azure Destination Storage Account and Application Registration, then configure an existing Harness Self-Managed Enterprise Edition deployment to utilize these components to act as a "staging area" for future Azure Billing Exports to be processed and made available for Harness CCM capabilities.
```

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 7]
```diff
@@ -10,31 +10,43 @@ redirect_from:
...
+
+**Figure: Azure CCM Self-Managed Documentation Goal**
+<DocImage path={require('./static/azure-smp-arch-initial-setup.jpeg')} title="Click to view full size image" />
+
+This goal will consist of four tasks. For Step 1, 2 and 3 please sign in to your [Azure Portal](https://portal.azure.com/#home):
```
Comment (brian-f-harness) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:22]:
```suggestion
This goal consists of four tasks. For Steps 1, 2, and 3 sign in to your [Azure Portal](https://portal.azure.com/#home): 
```

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 8]
```diff
@@ -10,31 +10,43 @@ redirect_from:
...
+
+This goal will consist of four tasks. For Step 1, 2 and 3 please sign in to your [Azure Portal](https://portal.azure.com/#home): 
 
 1. [Setup a new Application via App Registration](#setup-a-new-application-via-app-registration).
 2. [Create a new Client secret](#create-a-new-client-secret).
 3. [Create a new Destination Storage Account and a new Storage Container](#create-a-new-destination-storage-account-and-a-new-storage-container).
 4. [Deploy workloads via Helm charts](#deploy-workloads-via-helm-charts).
+
+:::info
+There are several times where we will ask you to note some items as suggested `variables`, please keep these values handy in a document or note of your choice as they will be used later in the process.
```
Comment (brian-f-harness) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:30]:
```suggestion
There are several times when you'll need to note some items as suggested `variables`. Keep these values in a document or note as they will be used later in the process.
```

**Review by brian-f-harness — CHANGES_REQUESTED** [review]

**Review by brian-f-harness — APPROVED** [review]
Approved through line 35. @joeyouss reviewing remaining changes for docs standards.

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 9]
```diff
@@ -10,31 +10,43 @@ redirect_from:
...
 ## Setup a new Application via App Registration
 
-1. Give it Name like `Harness CCM App`.
-2. For Supported account types select: `Accounts in any organizational directory (Any Microsoft Entra ID tenant - Multitenant)`.
-3. Click on Register. For more information, go to [Quickstart: Register an App](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app).
+Within your Microsoft Azure WebUI:
+1. Navigate to [App registrations](https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationsListBlade).
+2. Select "New Registration"
+3. Give your App registration a Name like `Harness CCM App`.
+4. For Supported account types select: `Accounts in any organizational directory (Any Microsoft Entra ID tenant - Multitenant)`.
```
Comment (joeyouss) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:39]: Under "Supported account types", select:

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 10]
```diff
@@ -10,31 +10,43 @@ redirect_from:
...
+5. Click on Register. For more information, go to [Quickstart: Register an App](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app).
```
Comment (joeyouss) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:40]: "Register"

Comment (joeyouss) [reply docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:40]: please make the options bold, whenever we are using them in our docs

**Review by brian-f-harness — COMMENTED** [review]
Jytoi reviewing, approving, and merging

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 11]
```diff
@@ -45,35 +57,33 @@ Copy & Save **Application (client) ID** → `clientId`.
...
-  <DocImage path={require('./static/azure-client-secret.png')} width="50%" height="50%" title="Click to view full size image" />
+  <DocImage path={require('./static/azure-client-secret.png')} title="Click to view full size image" />
 
-9. From the new Client secret, Copy & Save **Value** → `clientSecret`. For more information, go to [Add credentials](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app#add-credentials).
+9. From the new Client secret, Note & Save **Value** → `clientSecret`. For more information, go to [Add credentials](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app#add-credentials).
 
 :::info
 Make a note of the following:
-
 - tenantId
```
Comment (joeyouss) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:66]: please use backticks

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 12]
```diff
@@ -82,24 +92,21 @@ This will act as a staging area for CCM to sync data from multiple billing expor
...
 15. Click on **Generate SAS and connection string**. This is required to sync data from multiple source account to destination storage account.
-16. Save **SAS token** → `sasToken` starting from `sv=`, ignore `?` in beginning. For more information, go to [Create your SAS tokens](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/create-sas-tokens?view=doc-intel-4.0.0#use-azure-storage-explorer).
+16. Note **SAS token** as → `sasToken` starting from `sv=`, ignore `?` in beginning. For more information, go to [Create your SAS tokens](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/create-sas-tokens?view=doc-intel-4.0.0#use-azure-storage-explorer).
 
   <DocImage path={require('./static/azure-destination-storage-sas-token.png')} width="50%" height="50%" title="Click to view full size image" />
 
 :::info
 Make a note of the following:
-
 - storageName
```
Comment (joeyouss) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:101]: backticks

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 13]
```diff
@@ -82,24 +92,21 @@ This will act as a staging area for CCM to sync data from multiple billing expor
...
-
 - sasToken
 :::
 
 ## Deploy workloads via Helm charts
+From here, we will use the values we've noted from previous steps to update our Harness SMP deployment.
 
-1. Clone the chart repository.
-
+1. Clone the chart repository if you have not already.
```
Comment (joeyouss) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:109]: (if not done already)

File: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md [hunk 14]
```diff
@@ -156,44 +157,40 @@ helm upgrade <chart-name> <chart-directory> -n <namespace> -f override.yaml
 
 ## Handling Kubernetes secrets
 
-When installing or upgrading the Helm charts, Kubernetes secrets with default values are created within the cluster. These generated secrets should be updated with the values mentioned above. Before updating the secrets, you need to convert the secret into base64 encoded format. For example, if your **HARNESS_CE_AZURE_CLIENTID** value is "clientId", it would be stored as `Y2xpZW50SWQ==` after encoding. After changing secrets, kindly bounce the corresponding pods.
+When installing or upgrading the Helm charts, Kubernetes secrets with default values are created within the cluster. These generated secrets should be updated with the values mentioned above.
+
+ Before updating the secrets, you need to convert the secret into base64 encoded format. For example, if your **HARNESS_CE_AZURE_CLIENTID** value is "clientId", it would be stored as `Y2xpZW50SWQ==` after encoding. Please
```
Comment (joeyouss) [inline docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:162]: Base64

Comment (joeyouss) [reply docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md:162]: clientId should be in backticks

**Pull request merged**